lib/api/BackbeatServer.js

```diff
@@ -120,8 +122,19 @@ class BackbeatServer {
         && this._areConditionsOk(bbRequest)) {
             bbRequest.setStatusCode(200);
 
-            const routeDetails = routes.find(route =>
-                route.path === bbRequest.getRoute());
+            const route = bbRequest.getRoute();
+            let routeDetails = routes.find(r => r.path === route);
+            if (!routeDetails) {
+                // a valid route with an optional param
+                const subRoute = route.substr(0, route.lastIndexOf('/'));
+                routeDetails = routes.find(r => r.path === subRoute);
+
+                const site = route.substr(route.lastIndexOf('/') + 1);
```

Comment:
I think routes would be better organized (and also less prone to errors/name conflicts) in the following way (I'm not sure of the current exact semantics, so feel free to adapt to what is currently there, but I want to show the general idea).

```
/_/metrics/crr/all/backlog
/_/metrics/crr/all/throughput
...
/_/metrics/crr/aws_s3/throughput
```

Later we could also provide additional grouping capability like:
```
/_/metrics/by_site/aws_s3     // would provide all metrics related to aws_s3.
```

The first level names should be reserved as namespace, with a fixed number that we control in the implementation, avoiding the need to hammer in with some pain the new metrics later for various other components. Since no one has deployed or tooled this API yet, I think it's a good time to do this change.

Reply:
I'll be reflecting these changes in a new PR 👍 . For now, I'll build a new branch off this one so it will be easy to rebase later